airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java

```diff
@@ -39,96 +34,57 @@
-    this.sqlGenerator = sqlGenerator;
-    this.destinationHandler = destinationHandler;
+    this.typerDeduper = typerDeduper;
     this.catalog = catalog;
-    this.overwriteStreamsWithTmpTable = new HashMap<>();
-
     this.use1s1t = TypingAndDedupingFlag.isDestinationV2();
 
     LOGGER.info("Got parsed catalog {}", catalog);
     LOGGER.info("Got canonical stream IDs {}", uploaderMap.keySet());
   }
 
   @Override
-  protected void startTracked() throws InterruptedException {
+  protected void startTracked() throws Exception {
     // todo (cgardens) - move contents of #write into this method.
 
     if (use1s1t) {
-      // TODO extract common logic with GCS record consumer + extract into a higher level class
-      // For each stream, make sure that its corresponding final table exists.
-      for (final StreamConfig stream : catalog.streams()) {
-        final Optional<TableDefinition> existingTable = destinationHandler.findExistingTable(stream.id());
-        if (existingTable.isEmpty()) {
-          destinationHandler.execute(sqlGenerator.createTable(stream, ""));
-          if (stream.destinationSyncMode() == DestinationSyncMode.OVERWRITE) {
-            // We're creating this table for the first time. Write directly into it.
-            overwriteStreamsWithTmpTable.put(stream.id(), "");
-          }
-        } else {
-          destinationHandler.execute(sqlGenerator.alterTable(stream, existingTable.get()));
-          if (stream.destinationSyncMode() == DestinationSyncMode.OVERWRITE) {
-            final BigInteger rowsInFinalTable = bigquery.getTable(TableId.of(stream.id().finalNamespace(), stream.id().finalName())).getNumRows();
-            if (new BigInteger("0").equals(rowsInFinalTable)) {
-              // The table already exists but is empty. We'll load data incrementally.
-              // (this might be because the user ran a reset, which creates an empty table)
-              overwriteStreamsWithTmpTable.put(stream.id(), "");
-            } else {
-              // We're working with an existing table. Write into a tmp table. We'll overwrite the table at the
-              // end of the sync.
-              overwriteStreamsWithTmpTable.put(stream.id(), OVERWRITE_TABLE_SUFFIX);
-            }
-          }
-        }
-      }
+      typerDeduper.createFinalTables();
 
-      // For streams in overwrite mode, truncate the raw table and create a tmp table.
-      // non-1s1t syncs actually overwrite the raw table at the end of the sync, so we only do this in
-      // 1s1t mode.
-      for (final StreamConfig stream : catalog.streams()) {
-        LOGGER.info("Stream {} has sync mode {}", stream.id(), stream.destinationSyncMode());
-        final String suffix = overwriteStreamsWithTmpTable.get(stream.id());
-        if (stream.destinationSyncMode() == DestinationSyncMode.OVERWRITE && suffix != null && !suffix.isEmpty()) {
-          // drop+recreate the raw table
+      // Set up our raw tables
```

Comment:
I'm leaving the raw table setup out of the TyperDeduper for now. Standard inserts and staging are doing different things (e.g. staging also needs to set up staging areas, and standard inserts has some weird streaming upload thing objects to manage).

IMO we should revisit this after the async work is done, but I could be convinced to do a speculative refactor right now.